### PR TITLE
Issue 932: Prevent Dependabot version updates from running on HHS:main

### LINKED ISSUE #932: Prevent Dependabot version updates from running on HHS:main
**Description:**
Recently a `dependabot.yml` file was added to this repo to configure Dependabot to automatically issue pull requests nightly for any dependencies listed in package.json or Pipfile. However, to keep with our current git flow we don't want these pull requests to be issued against the main HHS repository. Automatic security updates have already been disabled on the HHS repo but those work distinctly from version updates where the configuration file is the source of truth and as such those PRs still get opened.

To resolve this we can specify a [target_branch](https://docs.github.com/en/code-security/supply-chain-security/configuration-options-for-dependency-updates#target-branch) in the configuration file so these version updates only check and issue PRs against the `raft-tdp-main` branch. Since this branch doesn't exist in HHS the PRs will not get opened. 

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [x] Dependabot version update PRs do not get opened against HHS:main
- [x] Testing Checklist has been run and all tests pass
- [x] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [x] Add a `target_branch` configuration to the `dependabot.yml` configuration file
- [x] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

- [Note about impact of this change](https://docs.github.com/en/code-security/supply-chain-security/customizing-dependency-updates#impact-of-configuration-changes-on-security-updates)

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_

### PR DESCRIPTION
**Pull request by jtwillis92**
### Summary of Changes
Updates the Dependabot configuration to only check dependencies and issue fixes against the branch `raft-tdp-main`. As a result, since the HHS:main repo does not have this branch this will prevent Dependabot PRs from being opened upstream.

Additionally, the `labels` configuration was added to set the labels for PRs opened by Dependabot to `dependencies`, `frontend` or `backend` and `raft review` to match the labels used on other PRs.

Closes #932 

### How to Test
As with the [previous Dependabot PR](#917) this change relates to built in functionality of GitHub so our only option to test before merge is to validate the config file by pasting it [here](https://dependabot.com/docs/config-file/validator/)

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  - [x] Dependabot version update PRs do not get opened against HHS:main

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

N/A no code changes

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

N/A no code changes

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

N/A no UX changes

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging

N/A no changes to deployed sites

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

N/A no code changes

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?

### CODE REVIEW

**Review by jorgegonzalez — APPROVED** [review]
LGTM

**Comment by ADPennington** [comment]
@jtwillis92 could you add an ADR to this PR? Template below, please feel free to edit, as needed. 

---

# 16. Migrating to Github Native Dependabot for TDP Dependency Management

Date: 2021-05-14 (yyyy-mm-dd)

## Status

Pending

## Context

Currently our Snyk configuration is configured on the `carltonsmith` Snyk organization which will no longer work after @carltonsmith leaves the project. Additionally, these Snyk PRs require us to manage an unnecessary `requirements.txt` file in addition to our Pipfile and python dependency update PRs are not complete when they get opened since they don't update anything in our actual dependencies. [Example Snyk Python Dependency PR](https://github.com/raft-tech/TANF-app/pull/903)

Additionally, we currently use the Dependabot Preview app which is being deprecated in favor of a [GitHub Native Dependabot](https://github.blog/2021-04-29-goodbye-dependabot-preview-hello-dependabot/) which has more features and is configured via a YAML file committed to the repo.

## Proposed Decision

Rather than setting up Snyk on a new organization and in order to get ahead on the impending Dependabot migration,  We propose an update which provides the necessary YAML config to enable the new GitHub Native version of Dependabot. 

## Consequences

To maintain our gitflow:
- We need to explicitly disable automated PR updates on the HHS repo if it is not already, otherwise once this file gets merged in we will get Dependabot security-related PRs to both repos. This can be disabled [in the security analysis settings](https://github.com/raft-tech/TANF-app/settings/security_analysis) as shown below:
disable-dependabot-security-updates
- We also need to specify a `target_branch` in the dependabot configuration file to prevent automated [version-related](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/about-dependabot-version-updates) PR updates from opening in the HHS repo. 

### Pros

- Mitigate the need to manage unnecessary file for dependency management
- Mitigate the need to manage multiple dependency management tools
- Dependency management no longer contingent upon project user accounts. 

### Cons

- None that we are aware of at this time. 

## Notes

Time estimate for development: n/a. #917 and #932 were created to support this migration and PR #944 complete this migration.

**Comment by jtwillis92** [comment]
> @jtwillis92 could you add an ADR to this PR? Template below, please feel free to edit, as needed.

Completed! Thank you for providing the template :raised_hands:

File: docs/Architecture Decision Record/016-dependabot-dependency-management.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# 16. Migrating to Github Native Dependabot for TDP Dependency Management
+Date: 2021-05-14 (yyyy-mm-dd)
+
+## Status
+
+Pending
```
Comment (ADPennington) [inline docs/Architecture Decision Record/016-dependabot-dependency-management.md:6]:
```suggestion
Approved
```

**Review by ADPennington — APPROVED** [review]
### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  + [x] Dependabot version update PRs do not get opened against HHS:main
_confirmed security-related PRs do not open in hhs repo. to validate version-related PRs don't open upon merge to `hhs:main`_ 
As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

_n/a_

### Deliverable 3: Properly Styled Code

_n/a_

### Deliverable 4: Accessible

_n/a_

### Deliverable 5: Deployed

_n/a_

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

_no code changes but ADR 016 includes full documentation trail for this change._

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
_none detected_

**Pull request merged**